src/lib.rs

```diff
@@ -66,3 +66,23 @@ pub mod x86;
 pub mod x86_64;
 
 pub mod probestack;
+
+/// Changes the visibility to `pub` if feature "public-test-deps" is set
+#[doc(hidden)]
+#[cfg(not(feature = "public-test-deps"))]
+#[macro_export]
+macro_rules! public_test_dep {
+    ($(#[$($meta:meta)*])* pub(crate) $ident:ident $($tokens:tt)*) => {
+        $(#[$($meta)*])* pub(crate) $ident $($tokens)*
+    };
+}
+
+/// Changes the visibility to `pub` if feature "public-test-deps" is set
+#[doc(hidden)]
+#[cfg(feature = "public-test-deps")]
+#[macro_export]
+macro_rules! public_test_dep {
+    {$(#[$($meta:meta)*])* pub(crate) $ident:ident $($tokens:tt)*} => {
+        $(#[$($meta)*])* pub $ident $($tokens)*
+    };
+}
```

Comment:
If you place the macros before the `mod` items, you shouldn't need `#[macro_export]` and `use crate::public_test_dep`.

Reply:
Thanks, I managed to remove all macros from the public interface. I'm pretty sure I tried this before and it wouldn't work, I must have had some detail wrong.